enhancements/20220214-harvester-cloud-provider-enhancement.md

```diff
@@ -0,0 +1,237 @@
+# Harvester Cloud Provider Enhancement
+
+## Summary
+
+Currently, the load balancer provided by the Harvester cloud provider is exposed in the local network of the Harvester hosts. If users in the VLAN network wants to connect the load balancer IP(VIP), they must prove the network route connectivity by themselves..
+
+There is another limitation about IPAM. Only a single CIDR range can be configured per Harvester namespace.
+
+To meet the requirement and provide a better user experience, we will deploy the kube-vip alone with the Harvester CCM and add the IPAM module in the Harvester load balancer controller to replace the kube-vip cloud provider.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/1762
+
+https://github.com/harvester/harvester/issues/1580
+
+## Motivation
+
+### Goals
+
+- Expose load balancer IP in the VLAN network of the guest cluster.
+- Support for configuring an IP range per VLAN network and an IP range per tenant.
+- Provide LB for traditional VM workloads deployed in the Harvester VMs.
+
+## Proposal
+
+### User Stories
+#### Story 1
+
+Before this enhancement, the VIP for the load balancer service of the guest cluster is only exposed in the Harvester LAN. However, in the most of real environment, users want to access the VIP in the VLAN where the guest clusters run. They have to prove the route connectivity by themselves which is a difficult stuff for those who are not familiar with the layer 3 networks.
+After this enhancement, the VIP can be exposed to the VLAN. Hence, users can access the VIP directly without any additional network configurations.
+
+#### Story 2
+
+Before this enhancement, an IP range can be only configured per Harvester namespace. After this enhancement, users can specify the IP range for the specified VLAN, projects, namespaces or guest clusters.
+
+#### Story 3
+If users want to distribute application traffic across the workloads deployed in the VMs rather than the guest cluster, they can create a LB of VM workload type and group the backend VMs by labels.
+
+### User Experience In Detail
```

Comment:
Harveter has a concept of `addon`, which is about enable/disable some features on the fly.
https://docs.harvesterhci.io/v1.1/advanced/addons

What is the meaning of `Add-on Config` here?

Reply:
It's just a tab name when creating the load balancer service in the guest cluster with Harvester cloud provider.